<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>572</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ImageRotator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Image:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="pathBox"/>
      </item>
      <item>
       <widget class="QPushButton" name="browseBtn">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="loadBtn">
        <property name="text">
         <string>Load</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="fitToWindowBtn">
        <property name="text">
         <string>Fit to window</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="resetZoomBtn">
        <property name="text">
         <string>Reset zoom</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Degrees to rotate:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="degBox">
        <property name="minimum">
         <number>-999999</number>
        </property>
        <property name="maximum">
         <number>999999</number>
        </property>
        <property name="singleStep">
         <number>15</number>
        </property>
        <property name="value">
         <number>45</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="rotateBtn">
        <property name="text">
         <string>Rotate</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="rotate45DegLeftBtn">
        <property name="text">
         <string>Rotate 45° left</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="rotate45DegRightBtn">
        <property name="text">
         <string>Rotate 45° right</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="flipVerticallyBtn">
        <property name="toolTip">
         <string>Note: will discard the current rotation</string>
        </property>
        <property name="text">
         <string>Flip original vertically</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="flipHorizontallyBtn">
        <property name="toolTip">
         <string>Note: will discard the current rotation</string>
        </property>
        <property name="text">
         <string>Flip original horizontally</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="resetBtn">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="saveAsBtn">
        <property name="text">
         <string>Save as...</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Interpolation method to use:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="methodBox">
        <item>
         <property name="text">
          <string>Nearest neighbor</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Bilinear</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="GraphicsViewEx" name="graphicsView"/>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GraphicsViewEx</class>
   <extends>QGraphicsView</extends>
   <header>graphicsviewex.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
